website/content/docs/security/acl/auth-methods/aws-iam.mdx

```diff
@@ -0,0 +1,199 @@
+      "MaxRetries": 3,
+      "IAMEndpoint": "https://iam.amazonaws.com/",
+      "STSEndpoint": "https://sts.us-east-1.amazonaws.com/",
+      "AllowedSTSHeaderValues": ["X-Extra-Header"]
+    }
+}
+```
+
+## Trusted Identity Attributes
+
+The authentication step returns the following trusted identity attributes for use in binding rule
+selectors and bind name interpolation. All of these attributes are strings that can be interpolated
+and support the following selector operations: `Equal, Not Equal, In, Not In, Matches, Not Matches`
+
+| Attribute            | Description                         | Requirement                                                      |
+| -------------------- | ----------------------------------- | ---------------------------------------------------------------- |
+| `entity_name`        | Name of IAM role or user            |                                                                  |
+| `entity_id`          | Unique ID of IAM role or user       |                                                                  |
+| `account_id`         | AWS account id of IAM role or user  |                                                                  |
+| `entity_path`        | The path of the IAM role or user    | `EnableIAMEntityDetails=true`                                    |
+| `entity_tags.<key>`  | AWS account id of IAM role or user  | `EnableIAMEntityDetails=true` and `IAMEntityTags` contains `<key>` |
+
+## IAM Policies
+
+When `EnableIAMEntityDetails=false`, no specific IAM policies are needed.
+
+When `EnableIAMEntityDetails=true`, an authenticating client must provide either a signed
+`iam:GetRole` or `iam:GetUser` request. This request is signed with the client's AWS credentials, so
+the client must have permission to fetch the role or user, respectively.
+
+- If the authenticating client is an IAM role, the client must have an `iam:GetRole` permission to
+  fetch its own role. The following shows an example of an IAM Policy document which grants this
+  permission:
+
+  ```json
+  {
+    "Statement": [
+        {
+            "Action": ["iam:GetRole"],
+            "Effect": "Allow",
+            "Resource": ["arn:aws:iam::123456789012:role/MyRoleName"]
+        }
+    ],
+    "Version": "2012-10-17"
+  }
+  ```
+
+- If the authenticating client is an IAM user, the client must have an `iam:GetUser` permission to
+  fetch its own role. The following shows an example of an IAM Policy document which grants this
+  permission:
+
+  ```json
+  {
+    "Statement": [
+        {
+            "Action": ["iam:GetUser"],
+            "Effect": "Allow",
+            "Resource": ["arn:aws:iam::123456789012:user/MyUserName"]
+        }
+    ],
+    "Version": "2012-10-17"
+  }
+  ```
+
+## Authentication Procedure
+
+If `EnableIAMEntityDetails=false`, a client must log in with `consul login -type aws-iam
+-aws-auto-bearer-token ...`. This command does the following:
```

Comment:
```suggestion
If `EnableIAMEntityDetails=false`, a client must log in with the `consul login` command. 

```shell-session
$ consul login -type aws-iam -aws-auto-bearer-token
```

This command does the following:
```

We when specifying commands we should make use of the shell-session component. That way, the user can easily find the command and see how it's used. The inline form in the sentence is easy to miss when scanning the document. So to help this pop, let's use the shell-session.